src/main/java/org/jenkinsci/test/acceptance/junit/WithPlugins.java

```diff
@@ -88,57 +90,66 @@ public Statement apply(final Statement base, final Description d) {
                 @Override
                 public void evaluate() throws Throwable {
                     jenkins = injector.getInstance(Jenkins.class);
-                    Set<String> plugins = new TreeSet<>();
-                    boolean restartRequired = installPlugins(d.getAnnotation(WithPlugins.class), plugins);
-                    restartRequired |= installPlugins(d.getTestClass().getAnnotation(WithPlugins.class), plugins);
-                    LOGGER.info("for " + d + " asked to install " + plugins + "; restartRequired? " + restartRequired);
-                    if (restartRequired) {
-                        assumeTrue("This test requires a restartable Jenkins", jenkins.canRestart());
-                        jenkins.restart();
-                    }
-                    for (String name : plugins) {
-                        Plugin installedPlugin = jenkins.getPlugin(name);
-                        VersionNumber installedVersion = installedPlugin.getVersion();
-                        String version = installedVersion.toString();
-                        pluginReporter.log(d.getClassName() + "." + d.getMethodName(), name, version);
+                    List<PluginSpec> plugins = combinePlugins(
+                            d.getAnnotation(WithPlugins.class),
+                            d.getTestClass().getAnnotation(WithPlugins.class)
+                    );
+
+                    installPlugins(plugins);
+
+                    for (PluginSpec plugin : plugins) {
+                        pluginReporter.log(
+                                d.getClassName() + "." + d.getMethodName(),
+                                plugin.getName(),
+                                plugin.getVersion()
```

Comment:
This is wrong. It just spits back out the contents of the `@WithPlugins` annotation, regardless of what Jenkins is actually running.